src/error.rs

```diff
@@ -171,21 +159,91 @@ impl ErrorReason {
 /// Returned when the given data slice is not properly aligned for the target
 /// type. It would have been properly aligned if `offset` bytes were shifted
 /// (discarded) from the front of the slice.
-#[derive(Debug, Clone, Eq, Hash, PartialEq)]
-pub struct UnalignedError {
+#[derive(Clone, Eq, Hash, PartialEq)]
+pub struct UnalignedError<'a, S, T> {
     /// The required amount of bytes to discard at the front for the attempted
     /// transmutation to be successful.
     pub offset: usize,
+    /// A slice to the original source data.
+    pub source: &'a [S],
+    phantom: PhantomData<T>,
+}
+
+impl<'a, S, T> UnalignedError<'a, S, T> {
+    pub fn new(offset: usize, source: &'a [S]) -> Self {
+        UnalignedError {
+            offset,
+            source,
+            phantom: PhantomData,
+        }
+    }
+
+    /// Create a copy of the source data, transmuted into a vector. As the new
```

Comment:
Why not "Create a copy of the data and transmute it." from below? Is there a difference I'm not seeing? (Also probably extract into own funxion maybe?)

Reply:
Meh, it feels strange for me to say "copy the data and then transmute it", since the transmutation is an implicit consequence of copying the data. I have chosen to make the docs of each function consistent.

And yes, having a helper function is a good idea.